tinygrad/shape/shapetracker.py

```diff
@@ -228,21 +228,21 @@ def expand(self, new_shape: Tuple[Union[Node,int], ...]) -> ShapeTracker:
     return self
 
   def reshape(self, new_shape: Tuple[Union[Node,int], ...]):
-    new_ints, new_nodes = partition(new_shape, lambda s: isinstance(s, int))
-    if new_nodes and all(isinstance(s, int) for s in self.shape):
-      # reshape from all int shape into shape with a variable, update the variable value
-      assert len(new_nodes) == 1 and isinstance(new_nodes[0], Variable), "only support adding one Variable to the int shape"
-      new_var, new_val = new_nodes[0], prod(self.shape) // prod(new_ints)
-      if new_var not in self.var_vals:
-        assert new_var.min <= new_val <= new_var.max, f"variable value {new_val} out of range [{new_var.min}, {new_var.max}]"
-        self.var_vals[new_var] = new_val
-      else: assert self.var_vals[new_var] == new_val, f"value conflicts, was {self.var_vals[new_var]}, set to {new_val}"
-    elif not new_nodes: self.var_vals = {}
     if self.views[-1].shape == new_shape: return self
+    if not all(s.__class__ is int for s in new_shape):
+      if all(s.__class__ is int for s in self.shape):
+        new_ints, new_nodes = partition(new_shape, lambda x: x.__class__ is int)
```

Comment:
This is applying a second pass of `__class__` over new_shape in addition to L232. Would the old way with only one pass be faster? Nesting `if all(s.__class__ is int for s in self.shape):` should be fine and faster.

Reply:
This way is faster, the partition call is expensive.